src/ruby/tools/bin/protoc_grpc_ruby_plugin.rb

```diff
@@ -30,14 +30,7 @@
 
 require 'rbconfig'
 
-require_relative '../os_check'
-
-plugin_name = 'grpc_ruby_plugin' + case OS.os_name
-                                   when 'windows'
-                                     '.exe'
-                                   else
-                                     ''
-                                   end
+plugin_name = 'grpc_ruby_plugin' + RbConfig::CONFIG['EXEEXT']
 
 plugin_path = File.join(File.dirname(__FILE__),
                         RbConfig::CONFIG['host_cpu'] + '-' + OS.os_name,
```

Comment:
You're still using "OS.os_name" here, aren't you ? You can't just delete that function :-)